roku_dev_cli/roku_dev_cli.py

```diff
@@ -95,14 +104,201 @@ def deployZip(zipPath, ipAddress, user):
+
+                    # make an array of lines and mark the line with rendezvous as True
+                    filePath = start.filePath
+                    if filePath not in fileLineSections:
+                        fileLineSections[filePath] = {
+                            "rendezvousLines" : []
+                        }
+                    lineNumber = start.lineNumber
+                    rendezvousLines = fileLineSections[filePath]["rendezvousLines"]
+                    while lineNumber > len(rendezvousLines): # make sure it's big enough
+                        rendezvousLines.extend([False] * 100)
+                    rendezvousLines[lineNumber] = True
+
+            fd.write("=== Rendezvous by Frequency ===\n")
+            for pathAndLine,stats in sorted(rByPathAndLine.items(), key=lambda kv: kv[1]["count"], reverse=True):
+                averageDuration = stats["totalTime"] / stats["count"]
+                fd.write("%i - %s (avg. duration: %.3fs)\n" % (stats["count"], pathAndLine, averageDuration))
+
+            fd.write("\n")
+
+            fd.write("=== Rendezvous by Longest Single Duration ===\n")
+            for pathAndLine,stats in sorted(rByPathAndLine.items(), key=lambda kv: kv[1]["longestDuration"], reverse=True):
+                fd.write("%2.3fs - %s\n" % (stats["longestDuration"], pathAndLine))
+
+            fd.write("\n")
+
+            fd.write("=== Adjacent Rendezvous (opportunity to use `queueFields`?) ===\n")
+
+            for filePath,stats in fileLineSections.items():
+                rendezvousLines = stats["rendezvousLines"]
+
+                adjacentLines = []
+                for lineNumber in range(0, len(rendezvousLines)):
+                    if rendezvousLines[lineNumber]: # this line has a rendezvous, search adjacent lines
+                        for adjacentLineNumber in range(lineNumber-adjacentRange, lineNumber+adjacentRange+1):
+                            if adjacentLineNumber > -1 and adjacentLineNumber < len(rendezvousLines) and lineNumber != adjacentLineNumber:
+                                if rendezvousLines[adjacentLineNumber]:
+                                    adjacentLines.append(lineNumber)
+                                    break
+
+                # hyphenate ranges
+                ranges = self.hyphenateRanges(adjacentLines, adjacentRange)  
+
+                if len(ranges) > 0:
+                    fd.write("%s - lines adjacent to other rendezvous: %s\n" % (filePath, ", ".join(ranges)))
+
+    # takes a list of int-formatted line numbers and groups them into hyphenated ranges
+    def hyphenateRanges(self, adjacentLines, adjacentRange):
+        ranges = []
+        for adjacentLine in adjacentLines:
+            if len(ranges) == 0:
+                ranges.append("%s-%s" % (adjacentLine, adjacentLine))
+            else:
+                lastRange = ranges[-1]
+                firstLineNumber, lastLineNumber = lastRange.split("-") # get last number in last range
```

Comment:
Actually, maybe this is the psuedo-sorting bit that keeps things from getting wonky?